system/jlib/jmetrics.hpp

```diff
@@ -237,6 +236,101 @@ class CustomMetric : public MetricBase
 };
 
 
+template <typename T>
+struct BucketInfo
+{
+    std::string label;
+    T limit;
+};
+
+
+
+template <typename T>
+struct Bucket
+{
+    Bucket(const std::string &_label, T _limit) :
+        label{_label}, limit{_limit}, count{0} {}
+    std::string label;
+    T limit;
+    RelaxedAtomic<__uint64> count;
+};
+
+
+struct BucketValue
+{
+    BucketValue(const std::string &_label, __uint64 _count) : label(_label), count(_count) {}
+    std::string label;
+    __uint64 count = 0;
+};
+
+
+template <typename T>
+struct HistogramMetricValues
+{
+    std::vector<BucketValue> bucketValues;
+    __uint64 count;
+    T sum;
+};
+
+template<typename T>
+class jlib_decl HistogramMetric : public MetricBase
+{
+public:
+    HistogramMetric(const char *name, const char *desc, StatisticMeasure _units, const std::vector<BucketInfo<T>> &bucketDefs, const MetricMetaData &_metaData = MetricMetaData()) :
+        MetricBase(name, desc, MetricType::METRICS_HISTOGRAM, _units, _metaData)
+        {
+            for (auto &b: bucketDefs)
+            {
+                buckets.emplace_back(Bucket<T>{b.label, b.limit});
```

Comment:
If you're using emplace_back() I think you should be able to say

        buckets.emplace_back(b.label, b.limit);

Reply:
Have you tried again now it is no longer a template - it compiles for me.